miio/huizuo.py

```diff
@@ -147,7 +296,288 @@ def set_brightness(self, level):
+
+            return self.set_property("heat_level", heat_level)
+        else:
+            raise HuizuoException("Your device doesn't support a heat management")
+
+    @command(
+        default_output=format_output(
+            "\n",
+            "------------ Lamp parameters -----------\n"
+            "Power: {result.is_on}\n"
+            "Brightness: {result.brightness}\n"
+            "Color Temperature: {result.color_temp}\n"
+            "\n"
+            "---------- Heater parameters -----------\n"
+            "Heater power: {result.is_heater_on}\n"
+            "Heat level: {result.heat_level}\n"
+            "Heat fault code (0 means 'OK'): {result.heater_fault_code}\n",
+        ),
+    )
+    def status(self) -> HuizuoStatus:
+        """Retrieve properties."""
+
+        return HuizuoStatus(
+            {
+                prop["did"]: prop["value"] if prop["code"] == 0 else None
+                for prop in self.get_properties_for_mapping()
+            }
+        )
+
+
+class HuizuoLampScene(Huizuo):
+    """Support for Huizuo Lamps with additional scene commands"""
+
+    # The next section contains the scene management commands
+    # Right now I have no devices with the scenes for live testing, so the following section
+    # generated based on device specitifations
+
+    @command(
+        default_output=format_output("On/Off switch"),
+    )
+    def scene_on_off(self):
+        """Switch the on/off (only for models with scenes support)."""
+        if self.model in MODELS_WITH_SCENES:
+            return self.set_property("on_off", 0)
+        else:
+            raise HuizuoException("Your device doesn't support scenes")
+
+    @command(
+        default_output=format_output("Add the brightness"),
+    )
+    def brightness_add(self):
+        """Add the brightness (only for models with scenes support)."""
+        if self.model in MODELS_WITH_SCENES:
+            return self.set_property("brightness_add", 0)
+        else:
+            raise HuizuoException("Your device doesn't support scenes")
+
+    @command(
+        default_output=format_output("Decrease the brightness"),
+    )
+    def brightness_dec(self):
+        """Decrease the brightness (only for models with scenes support)."""
+        if self.model in MODELS_WITH_SCENES:
+            return self.set_property("brightness_dec", 0)
+        else:
+            raise HuizuoException("Your device doesn't support scenes")
+
+    @command(
+        default_output=format_output("Switch between the brightnesses"),
+    )
+    def brightness_switch(self):
+        """Switch between the brightnesses (only for models with scenes support)."""
+        if self.model in MODELS_WITH_SCENES:
+            return self.set_property("brightness_switch", 0)
+        else:
+            raise HuizuoException("Your device doesn't support scenes")
+
+    @command(
+        default_output=format_output("Add the color temperature"),
+    )
+    def colortemp_add(self):
```

Comment:
See my comment wrt add/increase above. The method/property naming does not necessarily need to follow the "raw" property names where better/more comprehensible names are applicable.

Reply:
Done